### ISSUE TITLE
Upgrade commons-beanutils from 1.9.3 to 1.9.4

### ISSUE THREAD

**Issue by carlspring** [strongbox/strongbox#1847]
# Task Description

We need to upgrade `commons-beanutils:commons-beanutils:1.9.3` to `1.9.4`, as it fixes security vulnerabilites.

An automated pull request ( strongbox/strongbox-parent#78 )  was raised by `dependabot`.

# Tasks

The following tasks will need to be carried out:
* [ ] Update the version in the [strongbox-parent] project.
* [ ] Apply any required code changes, or dependency excludes.
* [ ] Test and confirm that there are no issues with the core projects ([strongbox], [strongbox-web-integration-tests], etc) after upgrading.
* [ ] Test against the `master` with the steps above.
* [ ] Test against the `issues/1649` with the steps above and create a pull request for for this against the `issues/1649` branch.

# Task Relationships

This task:
* Relates to: strongbox/strongbox-parent#78

# Help

* Check our article on [how to upgrade dependencies and plugins].
* [Our chat](https://chat.carlspring.org/)
* Points of contact:
  * @carlspring
  * @sbespalov
  * @steve-todorov

[strongbox]: https://github.com/strongbox/strongbox/
[strongbox-parent]: https://github.com/strongbox/strongbox-parent/
[strongbox-web-integration-tests]: https://github.com/strongbox/strongbox-web-integration-tests/
[how to upgrade dependencies and plugins]: https://strongbox.github.io/developer-guide/upgrading-dependencies-and-plugins.html

**Comment by ankitaugale23** [comment]
can I work on this?

**Comment by carlspring** [comment]
Hi @ankitaugale23 ,

Yes, please! Good luck!

**Comment by karannagi** [comment]
is this done or shall i work on it?

**Comment by carlspring** [comment]
Hi, @karannagi ,

Feel free to pick it up!

**Comment by kundra** [comment]
Hey! Is this still pending?

**Comment by carlspring** [comment]
Hi @kundra ,

Yes, it is! Feel free to pick it up! :)

**Comment by kundra** [comment]
Hey, @carlspring Please assign this to me.

**Comment by flaix** [comment]
Well, another week has passed and October is coming to an end. This is actually a dependency that should get updated due to existing security vulnerabilities in the current version. Just so you know, I have a PR on stand-by.

**Comment by carlspring** [comment]
Hi @flaix ,

Thanks for looking into this.

There is a problem here. @dependabot opened an automated pull request for this against the `master`. That worked back then (and needs to be re-tested). However, when @anki2189 was verifying this a while ago, this actuatlly caused some issue under the `issues/1649`. It's crucial that these upgrades don't break this branch, as we're doing some heavy development (we're migrating from OrientDB to Janusgraph + CassandraDB).

Would you like to re-check this against the `master` and `issues/1649`? We would really appreciate it! :)

**Comment by flaix** [comment]
It works against master, I checked that already. Will check `issues/1649`.

**Comment by carlspring** [comment]
Thanks! :)

**Comment by flaix** [comment]
Are the integration tests currently passing for `issues/1649`? They are failing for me. On the current branch tip, without the beanutils update. So I don't have a stable base to compare against.

Integration tests for Web Core are failing.
```
[INFO] Results:
[INFO]
[ERROR] Failures:
[ERROR]   NpmArtifactControllerTestIT.testSearchArtifactViaProxy:152 1 expectation failed.
JSON path objects.package.name doesn't match.
Expected: a collection containing "restonnode"
  Actual: []

[ERROR]   NpmArtifactControllerTestIT.testViewArtifactViaProxy:125 1 expectation failed.
JSON path versions.size() doesn't match.
Expected: a value greater than <0>
  Actual: 0

[ERROR]   RawArtifactControllerTestIT.testResolveArtifactViaProxy:54->testResolveArtifact:85->RawRestAssuredBaseTest.resolveArtifact:107 Failed to resolve /storages/storage0/ractit-raw-proxy/blob/master/ICLA.md!
[INFO]
[ERROR] Tests run: 225, Failures: 3, Errors: 0, Skipped: 0
```

**Comment by flaix** [comment]
A `mvn clean install` on `issues/1649` with the `commons-beanutils` updated to version 1.9.4 runs successfully.

Running `mvn clean install -Dintegration.tests` fails at the same tests as the current branch tip with version 1.9.3.

The usual test running Strongbox shows no errors.

Soo, the behaviour on branch `issues/1649` is the same with and without the update, but for my it has failing integration tests in both cases.

**Comment by carlspring** [comment]
@flaix ,

Could you please provide more details regarding the failures in the `strongbox-web-integration-tests`? Also, what is the complete output of your `mvn --version`?

Would you mind joining 
our [chat channel](https://chat.carlspring.org/), so we could discuss this furher?

@sbespalov : Are the tests passing for you from your branch?

**Comment by flaix** [comment]
I did not run the `strongbox-web-integration-tests`, I ran `mvn clean install -Dintegration.tests`. I'm not sure what details you are looking for?

```
$ mvn --version
Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
Maven home: /usr/local/maven
Java version: 1.8.0_272, vendor: AdoptOpenJDK, runtime: /Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home/jre
Default locale: en_DE, platform encoding: UTF-8
OS name: "mac os x", version: "10.14.6", arch: "x86_64", family: "mac"
```

**Comment by flaix** [comment]
I have run with `-Dintegration.tests` in the directory `strongbox-web-core`, but I don't think I can sensibly provide more detail from that than what is also available in the end result.

```
[ERROR] Tests run: 6, Failures: 2, Errors: 0, Skipped: 0, Time elapsed: 3.933 s <<< FAILURE! - in org.carlspring.strongbox.controllers.layout.npm.NpmArtifactControllerTestIT
[ERROR] testViewArtifactViaProxy{Repository}  Time elapsed: 0.588 s  <<< FAILURE!
java.lang.AssertionError:
1 expectation failed.
JSON path versions.size() doesn't match.
Expected: a value greater than <0>
  Actual: 0

	at org.carlspring.strongbox.controllers.layout.npm.NpmArtifactControllerTestIT.testViewArtifactViaProxy(NpmArtifactControllerTestIT.java:125)

[ERROR] testSearchArtifactViaProxy{Repository}  Time elapsed: 1.117 s  <<< FAILURE!
java.lang.AssertionError:
1 expectation failed.
JSON path objects.package.name doesn't match.
Expected: a collection containing "restonnode"
  Actual: []

	at org.carlspring.strongbox.controllers.layout.npm.NpmArtifactControllerTestIT.testSearchArtifactViaProxy(NpmArtifactControllerTestIT.java:152)
```

I'll leave you a PR, so you can verify against master that it is still running, if you like. And you could cherry pick the commit to test the `issues/1649` branch on your system, should the normal tip run there for you.

**Comment by carlspring** [comment]
If you could open one against the `issues/1649` branch, we could test that in Jenkins as well.

**Comment by flaix** [comment]
Same procedure with renaming the parent version and then referencing the changed parent version in strongbox/strongbox?

What would the branch names have to be for your automation to work?

**Comment by carlspring** [comment]
Only part of the CI build runs when a new pull request has been submitted, or a push notification has been recieved. If we see that passes, we trigged the whole build manually. We turn some very complex and expensive tests on our own infrastructure and it's not feasible to have this enabled for every commit/push, as it takes around 30 minutes in total.

The process is the same -- a pull for the `strongbox-parent` and one for the `strongbox` project itself.

**Comment by carlspring** [comment]
The complete build pipeline against the `master` [has pased](https://jenkins.carlspring.org/blue/organizations/jenkins/strongbox%2Fbuilds%2Fstrongbox/detail/PR-1951/2/pipeline).

**Comment by carlspring** [comment]
Also, regarding the branch names, it should either be `issue-1234`, or `issues/1234`. Otherwise, our Jenkins will not kick in.

**Comment by carlspring** [comment]
Thanks! You can use some custom names, but @steve-todorov can tell you more about that. We had it in the wiki somewhere.

**Comment by steve-todorov** [comment]
Hey, 
I've added some clarifications in the [docs](https://strongbox.github.io/developer-guide/upgrading-dependencies-and-plugins.html#1-strongbox-parent) regarding the versions. 
You are mixing naming conventions which is why it's not working.
I've fixed the `strongbox-parent` version for you, but you will need to fix the pom versions in strongbox/strongbox#1951 yourself so that it points to the proper `strongbox-parent` version.
